sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -205,7 +205,9 @@ def __init__(self, flags=None, **kwargs):
 
     # Build parser that will parse options recognized by the [sub]class of
     # PipelineOptions whose object is being instantiated.
-    parser = _BeamArgumentParser()
+    # set allow_abbrev=False to avoid prefix matching while parsing.
+    # https://docs.python.org/3/library/argparse.html#partial-parsing
+    parser = _BeamArgumentParser(allow_abbrev=False)
```

Comment:
This is to address ambiguous options `input` and `input_options`? Will this break users who have previously written pipelines with other abbreviations, esp custom ones?

Reply:
Yes, it is to address the two options. I am not sure about this breaking the users code(at least all the unit tests and IT tests passed in our case). @tvalentyn could add more perspective

Reply:
I'm also a little nervous about this change - could you help me understand why we need it?

It seems like it would definitely have the potential to break existing pipelines - if we need to do that, it might be better to detect and warn that the behavior will be changed in the future.